TIGLViewer/src/TIGLViewerSettings.cpp

```diff
@@ -54,6 +59,62 @@ void TIGLViewerSettings::setBGColor(const QColor& col)
     _bgcolor = col;
 }
 
+void TIGLViewerSettings::setShapeColor(const QColor& col)
+{
+    _shapecolor = col;
+}
+
+void TIGLViewerSettings::setShapeSymmetryColor(const QColor& col)
+{
+    _shapesymmetrycolor = col;
+}
+
+void TIGLViewerSettings::setDefaultMaterial(Graphic3d_NameOfMaterial material)
+{
+    _defaultMaterial = material;
+}
+
+Graphic3d_NameOfMaterial TIGLViewerSettings::defaultMaterial() const
+{
+    return _defaultMaterial;
+}
+
+void TIGLViewerSettings::setDefaultShapeColor(int r, int g, int b, int a)
+{
+    _shapecolor = QColor(r,g,b,a);
+    for (auto & listener : _settingsListener)
+    {
+        listener->defaultShapeColorHasChanged();
+    }
+}
+
+void TIGLViewerSettings::setDefaultShapeSymmetryColor(int r, int g, int b, int a)
+{
+    _shapesymmetrycolor = QColor(r,g,b,a);
+    for (auto & listener : _settingsListener)
+    {
+        listener->defaultShapeSymmetryColorHasChanged();
+    }
+}
+
+void TIGLViewerSettings::setBackgroundColor(int r, int g, int b, int a)
+{
+    _bgcolor = QColor(r,g,b,a);
+    for (auto & listener : _settingsListener)
+    {
+        listener->backgroundColorHasChanged();
+    }
+}
+
+void TIGLViewerSettings::setDefaultMaterial(const QString& material)
+{
+    _defaultMaterial = tiglMaterials::materialMap[material];
```

Comment:
Here, a check is required, if ``material`` is valid, i.e. if it's in the map. Otherwise an error should be printed to console.

Reply:
This has been resolved.